### PR TITLE
Update LICENSE and project metadata

### PR DESCRIPTION
**Pull request by sethmlarson**
@njsmith Let me know if something looks off. The current GitHub and Gitter tag-lines are not the same, I went with Gitter because it seemed better for me?

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 long_description = file: README.rst, CHANGES.rst
-keywords = urllib, httplib, threadsafe, filepost, http, https, ssl, pooling
-license = MIT
-author = Andrey Petrov
-author_email = andrey.petrov@shazow.net
-url = https://hip.readthedocs.io/
+keywords = http client
+license = MIT -or- Apache-2.0
+author = Nathaniel J. Smith
+author_email = njs@pobox.com
```
Comment (njsmith) [inline setup.cfg]: I feel a little weird claiming authorship here... I know normally the author field is like "yeah uh this is who first setup the package and they might or might not have anything to do with it anymore", but here I'm already probably not the biggest contributor at the time when we're setting the field :-)

Maybe we should leave this out, or put something like "The Hip project maintainers" or "The Trio Collective"?

Comment (pquentin) [reply setup.cfg]: +1 for "The Trio Collective". Do we need to keep the email?

Comment (RatanShreshtha) [reply setup.cfg]: +1 for "The Trio Collective".

Comment (sethmlarson) [reply setup.cfg]: I also like Trio Collective so I'll use that. For now I can keep your email address @njsmith as it is a "recommended" field.

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,3 @@
+This software is made available under the terms of *either* of the
+licenses found in LICENSE.APACHE2 or LICENSE.MIT. Contributions to
+Hip are made under the terms of *both* these licenses.
```
[the remark this answers is not in the document]
Comment (sethmlarson) [reply LICENSE]: Done!

**Comment by sethmlarson** [comment]
Looks like pypy3 is timing out? The failure is not related to this PR.

**Comment by pquentin** [comment]
Thanks!

**Pull request merged**